### PR TITLE
Modified POST function in code.py

### PR DESCRIPTION
**Pull request by mheiman**
Infogami's save_many method supports a data parameter, along with the query, comment, and action parameters, but the POST API defined here doesn't recognize the data parameter. This change makes the POST API consistent with the save_many parameters.

This change is necessary to implement Open Library work merge undo.

See https://github.com/internetarchive/openlibrary/pull/6948 for OL Side

### Stakeholders
@mheiman @cdrini

### CODE REVIEW

File: infogami/plugins/api/code.py [hunk 1]
```diff
@@ -103,7 +103,8 @@ def POST(self):
         h = get_custom_headers()
         comment = h.get('comment')
         action = h.get('action')
```
Comment (cclauss) [inline infogami/plugins/api/code.py:105]:
```suggestion
        data = {key: h.get(key) for key in ('action', 'comment', 'data')}
        data['query'] = query
```

Comment (mheiman) [reply infogami/plugins/api/code.py:105]: Nice cleanup -- thanks.

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
% `codespell`
```
./infogami/infobase/account.py:250: Lagacy ==> Legacy
./infogami/utils/markdown/markdown.py:646: Patter ==> Pattern
```

**Comment by cclauss** [comment]
If you rebase, the tests should pass.

**Review by cdrini — APPROVED** [review]
Tested:

```js
var r = await fetch('/works/OL53924W.json').then(r => r.json());
r.title += ' BLUE';

function save_many(items, comment, action, data) {
    console.log(`Saving ${items.length} items`);
    const headers = {
        Opt: '"http://openlibrary.org/dev/docs/api"; ns=42',
        '42-comment': comment,
        '42-action': action,
        '42-data': JSON.stringify(data),
    };

    return fetch('/api/save_many', {
        method: 'POST',
        headers,
        body: JSON.stringify(items),
    });
}

await save_many([r], 'Make it Blue', 'blue-it', {items: [1,2,3,4,4]})

function save_many(items, comment, action, data) {
    console.log(`Saving ${items.length} items`);
    const headers = {
        Opt: '"http://openlibrary.org/dev/docs/api"; ns=42',
        '42-comment': comment,
        '42-action': action,
    };

    return fetch('/api/save_many', {
        method: 'POST',
        headers,
        body: JSON.stringify(items),
    });
}

await save_many([r], 'Make it Blue', 'blue-it', {items: [1,2,3,4,4]})
```

And I verified http://localhost:8080/works/OL53924W.json?m=history

```json
[
    {
        "key": "/works/OL53924W",
        "revision": 3,
        "id": 51,
        "action": "blue-it",
        "author_id": 524,
        "ip": "172.29.0.1",
        "comment": "Make it Blue",
        "bot": false,
        "created": "2022-10-04 04:22:46.028576",
        "changes": "[{\"key\": \"/works/OL53924W\", \"revision\": 3}]",
        "data": "{}",
        "author": "/people/openlibrary"
    },
    {
        "key": "/works/OL53924W",
        "revision": 2,
        "id": 50,
        "action": "blue-it",
        "author_id": 524,
        "ip": "172.29.0.1",
        "comment": "Make it Blue",
        "bot": false,
        "created": "2022-10-04 04:20:25.907842",
        "changes": "[{\"key\": \"/works/OL53924W\", \"revision\": 2}]",
        "data": "{\"items\": [1, 2, 3, 4, 4]}",
        "author": "/people/openlibrary"
    },
    {
        "key": "/works/OL53924W",
        "revision": 1,
        "id": 42,
        "action": "bulk_update",
        "author_id": 24,
        "ip": "127.0.0.1",
        "comment": null,
        "bot": false,
        "created": "2013-03-28 07:50:48.835814",
        "changes": "[{\"key\": \"/works/OL16226033W\", \"revision\": 1}, {\"key\": \"/works/OL15547698W\", \"revision\": 1}, {\"key\": \"/works/OL15673999W\", \"revision\": 1}, {\"key\": \"/works/OL53924W\", \"revision\": 1}, {\"key\": \"/works/OL5702375W\", \"revision\": 1}, {\"key\": \"/works/OL192489W\", \"revision\": 1}, {\"key\": \"/works/OL6037022W\", \"revision\": 1}, {\"key\": \"/works/OL15692545W\", \"revision\": 1}, {\"key\": \"/works/OL15437498W\", \"revision\": 1}, {\"key\": \"/works/OL8193488W\", \"revision\": 1}, {\"key\": \"/works/OL54120W\", \"revision\": 1}, {\"key\": \"/works/OL73243W\", \"revision\": 1}, {\"key\": \"/works/OL16075717W\", \"revision\": 1}, {\"key\": \"/works/OL1898308W\", \"revision\": 1}, {\"key\": \"/works/OL61982W\", \"revision\": 1}, {\"key\": \"/works/OL15298516W\", \"revision\": 1}, {\"key\": \"/works/OL20600W\", \"revision\": 1}, {\"key\": \"/works/OL45310W\", \"revision\": 1}, {\"key\": \"/works/OL15706726W\", \"revision\": 1}, {\"key\": \"/works/OL15400392W\", \"revision\": 1}, {\"key\": \"/works/OL13101191W\", \"revision\": 1}, {\"key\": \"/works/OL151880W\", \"revision\": 1}, {\"key\": \"/works/OL118420W\", \"revision\": 1}, {\"key\": \"/works/OL5688746W\", \"revision\": 1}]",
        "data": "{}",
        "author": "/people/admin"
    }
]
```

**Pull request merged**